### DISCUSSION TITLE
Callback when music has finished playing

### DISCUSSION THREAD

**Discussion by FunkyFab** [pschatzmann/ESP32-A2DP#195] [Q&A] [answered]
Currently, there's no way to know (as far as I know) when a2dp_source.write_data has finished playing the music.
That'd be great if we could have a callback to be called when the music is finished.

**Comment by pschatzmann** [comment, marked as answer]
Hmm, I am a little bit confused. 
I usually use this to stream music from Spotify and there is not really any end. 

I suggest that you have a look at the functions related to audio_state:
- get_audio_state
- set_audio_state_callback
